### PR TITLE
T18021 remove deprecation warning

### PR DESCRIPTION
**Pull request by mgalka**
### Switch from deprecated pymongo methods to new PyMongo API
    
Switched from collection.remove to collection.delete_many. The change of
the PyMongo API required also modification of lab and token DELETE
handlers.

### Remove DELETE handlers for build, job and boot documents.
    
There are no use cases for DELETING these documents, apart from that
deletion of boot and job documents with the current handlers might have
lead to data integrity issues. Surplus tests are removed, too.

### CODE REVIEW

**Comment by gctucker** [comment]
Tested OK on staging.

File: app/handlers/lab.py [hunk 1]
```diff
@@ -150,17 +150,17 @@ def execute_delete(self, *args, **kwargs):
 
         return response
 
-    def _delete(self, spec_or_id, **kwargs):
+    def _delete(self, id_, **kwargs):
```
Comment (gctucker) [inline app/handlers/lab.py]: Maybe call it `oid` rather than `id_`?

File: app/utils/db.py [hunk 2]
```diff
@@ -503,23 +504,34 @@ def find_and_update(collection, query, document, operation="$set"):
     return ret_val
 
 
-def delete(collection, spec_or_id):
-    """Remove a document or multiple documents from the collection.
+def delete_by_id(collection, id_):
```
Comment (gctucker) [inline app/utils/db.py]: Ditto - maybe `oid` would be better than `id_`.

**Comment by gctucker** [comment]
Tested OK on staging.

File: app/handlers/lab.py (hunk 1 above)
Comment (mgalka) [reply app/handlers/lab.py]: `oid` makes more sense. Corrected.

File: app/utils/db.py (hunk 2 above)
Comment (mgalka) [reply app/utils/db.py]: Corrected.

**Comment by gctucker** [comment]
Tested OK again on staging.

**Pull request merged**